prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1116 @@
+                                                           .map { FhirTranscoder.decode(it) }
+
+            // there should only be one observation of five remaining, and the code of that observation
+            // should be 94558-5
+            for (fhirBundleAsObject in fhirBundlesAsObjectsOnly) {
+                assertThat(fhirBundleAsObject.getObservations().size == 1)
+                assertThat(fhirBundleAsObject.getObservations()[0].code.coding.size == 1)
+                assertThat(fhirBundleAsObject.getObservations()[0].code.coding[0].code.equals("94558-5"))
+            }
+
+            // is the queue messaging what we expect it to be?
+            val expectedRouteQueueMessages = reportAndBundles.flatMap { (report, fhirBundle) ->
+                listOf(
+                    FhirTranslateQueueMessage(
+                        report.reportId,
+                        report.bodyUrl,
+                        BlobAccess.digestToString(BlobAccess.sha256Digest(fhirBundle)),
+                        "phd.fhir-elr-no-transform",
+                        fhirSenderWithNoTransform.topic,
+                        "phd.x"
+                    ),
+                    FhirTranslateQueueMessage(
+                        report.reportId,
+                        report.bodyUrl,
+                        BlobAccess.digestToString(BlobAccess.sha256Digest(fhirBundle)),
+                        "phd.fhir-elr-no-transform",
+                        fhirSenderWithNoTransform.topic,
+                        "phd.y"
+                    )
+                )
+            }.map {
+                it.serialize()
+            }
+
+            verify(exactly = 2) {
+                QueueAccess.sendMessage(
+                    elrTranslationQueueName,
+                    match {
+                        expectedRouteQueueMessages.contains(it)
+                    }
+                )
+            }
+
+            // make sure action table has a new entry
+            checkActionTable(listOf(TaskAction.convert, TaskAction.receive, TaskAction.route))
+        }
+    }
+
+    @Test
+    fun `should respect jurisdictional filter and send message`() {
+        // set up
+        val reportContents =
+            listOf(
+                File(VALID_FHIR_URL).readText()
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val receiveReport = reportPair.first
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        val receivers = createReceivers(listOf(ReceiverSetupData("x", jurisdictionalFilter = jurisdictionalFilterTx)))
```

Comment:
See my comment above regarding default values for filters. Theoretically, this test should pass if the other filters are not set to true. The biggest issue I have with this setup is I don't think filters actually default to true if they are not specified, which they often are not in production.